### PR TITLE
JBPM-4953,JBPM-4303 - Create jBPM implementations of the WorkItemManager

### PR DESCRIPTION
**Pull request by mrietveld**

### CODE REVIEW

**Comment by mrietveld** [comment]
@mariofusco , @mswiderski  Would you mind reviewing this? 

I will be offline from the 1rst to the 8th, but will respond after that!

**Comment by MarianMacik** [comment]
+1, nice refactoring there!

File: drools-core/src/main/java/org/drools/core/event/ProcessEventSupport.java [hunk 1]
```diff
@@ -156,6 +156,18 @@ public void fireAfterVariableChanged(final String name, final String id,
         }
     }
 
+    public void fireBeforeNodeRemoved(final NodeInstance nodeInstance, KnowledgeRuntime kruntime) {
```
Comment (mswiderski) [inline drools-core/src/main/java/org/drools/core/event/ProcessEventSupport.java:159]: to follow comment in droolsjbpm-knowledge, I'd like to see the fireAfterNodeRemoved as well

File: drools-core/src/test/java/org/drools/core/process/instance/WorkItemManagerTest.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
+package org.drools.core.process.instance;
```
Comment (mswiderski) [inline drools-core/src/test/java/org/drools/core/process/instance/WorkItemManagerTest.java:1]: missing license header

File: drools-persistence-jpa/src/test/java/org/drools/persistence/jpa/processinstance/JPAWorkItemManagerTest.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
+package org.drools.persistence.jpa.processinstance;
```
Comment (mswiderski) [inline drools-persistence-jpa/src/test/java/org/drools/persistence/jpa/processinstance/JPAWorkItemManagerTest.java:1]: missing license header

**Comment by mswiderski** [comment]
just few minor comment (most imporatn about fireAfterNodeRemove event support), other than that looks ok

**Pull request closed without merging**